src/common/redis_module/ray_redis_module.cc

```diff
@@ -562,46 +598,48 @@ int PublishTableAdd(RedisModuleCtx *ctx,
   return RedisModule_ReplyWithSimpleString(ctx, "OK");
 }
 
-/// Add an entry at a key. This overwrites any existing data at the key.
-/// Publishes a notification about the update to all subscribers, if a pubsub
-/// channel is provided.
-///
-/// This is called from a client with the command:
-//
-///    RAY.TABLE_ADD <table_prefix> <pubsub_channel> <id> <data>
-///
-/// \param table_prefix The prefix string for keys in this table.
-/// \param pubsub_channel The pubsub channel name that notifications for
-///        this key should be published to. When publishing to a specific
-///        client, the channel name should be <pubsub_channel>:<client_id>.
-/// \param id The ID of the key to set.
-/// \param data The data to insert at the key.
-/// \return The current value at the key, or OK if there is no value.
-int TableAdd_RedisCommand(RedisModuleCtx *ctx,
-                          RedisModuleString **argv,
-                          int argc) {
+// RAY.TABLE_ADD:
+//   TableAdd_RedisCommand: the actual command handler.
+//   (helper) TableAdd_DoWrite: performs the write to redis state.
+//   (helper) TableAdd_DoPublish: performs a publish after the write.
+//   ChainTableAdd_RedisCommand: the same command, chain-enabled.
+
+int TableAdd_DoWrite(RedisModuleCtx *ctx,
```

Comment:
We recently turned on automatic Redis memory management in https://github.com/ray-project/ray/commit/9b9ff19dd087251c777df8d88a529ab951bc6eb3. Can you make sure all of the Redis commands added in this PR do this properly? That is, put `RedisModule_AutoMemory(ctx);` at the beginning of each command.